### ISSUE TITLE
The first search (after application start) hangs for a few seconds

### ISSUE THREAD

**Issue by fabchelly** [Flow-Launcher/Flow.Launcher#1201]
The first query after Flow Launcher starts is always extremely slow (whatever I'm trying to search) like if some index is created at first search.
It's probably triggered by one of the plugins using "*" as pattern (it hangs after the first character entered whatever the query is). After this first search every following request behaves normally.

In order to prevent this first slow-down, can a first search be simulated just after Flow Launcher starts?
It should be easy to simulate a search at startup (with a random string of 1 or 2 chars) to trigger this slow indexing process (for any plugin that need it the first time).

Thanks,
Fabio

**Comment by deefrawley** [comment]
The log should show the startup process and what is taking the longest. If you type `open log` in Flow it should show the result to open the log file location.

**Comment by jjw24** [comment]
I think this might be the same issue with the discussion here https://github.com/Flow-Launcher/Flow.Launcher/discussions/586.

@fabchelly can you replicate this problem everytime flow starts? Can you attach the log here after it happened please.

**Comment by fabchelly** [comment]
Yes it happens everytime after start. I cleared the log and restarted the program. The log contained 629 lines after start. The problem is maybe related to the error displayed in the log starting at line 630 (Spotify?).
[2022-05-15.txt](https://github.com/Flow-Launcher/Flow.Launcher/files/8695483/2022-05-15.txt)

**Comment by fabchelly** [comment]
I tried different times to restart the app and there are other errors appearing in the logs:
[2022-05-15-error2.txt](https://github.com/Flow-Launcher/Flow.Launcher/files/8695515/2022-05-15-error2.txt)

**Comment by taooceros** [comment]
We are supposed to have a asynchronous access to icon so theoretically that shouldn't be an issue. However, probably it is stunt because of the buffer is not ready?

**Comment by fabchelly** [comment]
I have uninstalled SpotifyPremium and I don't have the error anymore but the first search stills hangs for 2 seconds (after 1 character entered) before displaying the result

**Comment by taooceros** [comment]
It may be a reasonably workaround. Though a bit unclean for implementation. But I think it is a UI issue. So you still need to wait for the simulated query freezes for a while?

Anyway we can take a try. I will give it a test and send you an artifact later.

**Comment by fabiochelly** [comment]
Yes, nothing has change (on both of my computers).
There are both very recent (i7 and i9 from this year) so I don't think it is related to a lack of CPU power.

It can be related to the graphics as both screens are 4K set to 300 dpi and there may be a latency to recalculate element on those definitions.

Whats happens precisely is that if I type "flow"  after a restart for instance, only the "f" appears.
Then, after a 2 to 5 seconds freeze, the rest of the word appears instantly with the response.

**Comment by taooceros** [comment]
yeah I understand your issue. We have seen that before, but I have no idea how to fix it since it is very likely a UI framework issue.

**Comment by fabchelly** [comment]
> Anyway we can take a try. I will give it a test and send you an artifact later.

I'm ready to test the workaround as soon as it is ready. Thanks.

**Comment by Jack251970** [comment]
It should be fixed in the latest version. Let me know if your guys want to reopen this issue.
